client/state/data-layer/wpcom/videos/poster/index.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { VIDEO_EDITOR_UPDATE_POSTER } from 'state/action-types';
+import {
+	setPosterUrl,
+	showError,
+	showUploadProgress,
+} from 'state/ui/editor/video-editor/actions';
+
+/**
+ * Updates the poster for a video.
+ *
+ * @param  {Object} store  Redux store
+ * @param  {Object} action  Action object
+ * @param {Function} next  Dispatches to next middleware in chain
+ * @returns {Object} original action
+ */
+export const updatePoster = ( { dispatch }, action, next ) => {
+	if ( ! ( 'file' in action.params ) && ! ( 'at_time' in action.params ) ) {
+		return next( action );
+	}
+
+	const params = {
+		apiVersion: '1.1',
+		method: 'POST',
+		path: `/videos/${ action.videoId }/poster`,
+	};
+
+	if ( 'file' in action.params ) {
+		params.formData = [ [ 'poster', action.params.file ] ];
+	}
+
+	if ( 'at_time' in action.params ) {
+		params.body = action.params;
+	}
```

Comment:
if we use `Object.assign()` we can merge these imperative assignments into one declarative list and get rid of those `if` structures

```js
const params = Object.assign(
	{
		apiVersion: '1.1',
		method: 'POST',
		path: …
	},
	file && { formData: [ [ 'poster', file ] ] },
	at_time && { body: action.params },
);
```